msticpy/data/uploaders/splunk_uploader.py

```diff
@@ -97,27 +99,35 @@ def _post_data(
                 title="Splunk host not connected",
             )
         if not host:
-            host = "Upload"
+            host = "msticpy_splunk_uploader"
         index = self._load_index(index_name, kwargs.get("create_index", False))
         progress = tqdm(total=len(data.index), desc="Rows", position=0)
+        source_types = []
         for row in data.iterrows():
-            data = row[1].to_csv()  # type: ignore
+            if source_type == "json":
+                data = row[1].to_json()  # type: ignore
+            elif source_type == "csv":
+                data = row[1].to_csv()  # type: ignore
+            else:
+                data = row[1].to_string()  # type: ignore
             try:
                 data.encode(encoding="latin-1")  # type: ignore
             except UnicodeEncodeError:
                 data = data.encode(encoding="utf-8")  # type: ignore
-            index.submit(data, sourcetype=table_name, host=host)
+            index.submit(data, sourcetype=source_type, host=host)
+            source_types.append(source_type)
             progress.update(1)
         progress.close()
         if self._debug is True:
-            print("Upload complete")
+            print(f"Upload complete: Splunk sourcetype = {source_types}")
 
     # pylint: disable=arguments-differ
     def upload_df(  # type: ignore
         self,
-        data: pd.DataFrame,
```

Comment:
Would be good not to change the order of existing parameters - this is what is breaking tests.
I always add new parameters to the end of the existing named parameters  - that way it won't break existing cases that are using positional parameters.

Reply:
Thanks very much for your advice!